src/lycophron/tasks/tasks.py

```diff
@@ -6,218 +6,143 @@
 # under the terms of the MIT License; see LICENSE file for more details.
 """Lycophron tasks implementation."""
 
-import json
-import os
-import time
-from celery import Task
+from datetime import datetime
+from time import sleep
+import traceback
+
+from inveniordm_py.files.metadata import FilesListMetadata, OutgoingStream
+from inveniordm_py.records.metadata import DraftMetadata
+from requests.exceptions import HTTPError
 
-from ..client import create_session
 from ..db import LycophronDB
-from ..models import RecordStatus
+from ..errors import ErrorHandler
+from ..models import FileStatus, RecordStatus
 from . import app
 
-# TODO tasks are using 'db' directly. Breaks the basic flow Interface -> Business -> Data.
-# TODO tasks do not have any logging implemented yet (e.g. success, errors).
-# TODO record serialization to zenodo is done in place, should have its own module (e.g. marshmallow serializers)
-
-# TODO db instantiation needed here to avoid a MemoryError
-db_uri = "sqlite:///lycophron.db"
-db = LycophronDB(uri=db_uri)
-
-# TODO can be moved to configs
-RATE_LIMIT = 10
-SLEEP_TIME_SECONDS = 5
-
-
-class SqlAlchemyTask(Task):
-    """An abstract Celery Task that ensures that the connection the the
-    database is closed on task completion"""
-
-    abstract = True
-
-    def after_return(self, status, retval, task_id, args, kwargs, einfo):
-        db.session.remove()
 
-
-@app.task(base=SqlAlchemyTask)
-def publish_records(url, token, num_records=None):
+def state_transition(frm, to):
+    def update_state(event):
+        def wrapper(client, obj, *args, **kwargs):
+            from lycophron.app import app as lapp
```

Comment:
Minor: Connected to my previous comments. I think we should be consistent how we instantiate the Lycophron App, I see that we started using the singleton approach, so maybe we could stick to it, plus I would avoid dynamic imports if possible
```suggestion
            lapp = LycophronApp()
```

Reply:
I agree, I'll switch to the other one everywhere. The dynamic import is to avoid circular import based on current design (subject to change in the future).